### ISSUE TITLE
Why does `python` command in xonsh virtualenv not use $PATH

### ISSUE THREAD

**Issue by mwm-twx** [xonsh/xonsh#3770]
I have installed xonsh in a virtual environment to avoid conflicting dependencies.  When I start `xonsh.exe` from the `Scripts` directory, the `python` command executes the virtual environment python executable, rather than the one on my path (and shown by `which`).

## xonfig

<details>

```
$ xonfig
+------------------+-------------------------------------------------------+
| xonsh            | 0.9.21                                                |
| Python           | 3.8.1                                                 |
| PLY              | 3.11                                                  |
| have readline    | False                                                 |
| prompt toolkit   | 3.0.7                                                 |
| shell type       | prompt_toolkit                                        |
| pygments         | 2.6.1                                                 |
| on posix         | False                                                 |
| on linux         | False                                                 |
| on darwin        | False                                                 |
| on windows       | True                                                  |
| on cygwin        | <xonsh.lazyasd.LazyBool object at 0x00000236BBD9D4F0> |
| on msys2         | <xonsh.lazyasd.LazyBool object at 0x00000236BBD9D550> |
| is superuser     | False                                                 |
| default encoding | utf-8                                                 |
| xonsh encoding   | utf-8                                                 |
| encoding errors  | surrogateescape                                       |
+------------------+-------------------------------------------------------+
```

</details>

## Expected Behavior
Python executable is determined by the path.

## Current Behavior
Python executable is from the xonsh virtual environment.  `$(which python) script.py` will use the system python.

## Steps to Reproduce
1. Install xonsh in virtual environment
2. Run `xonsh.exe` from `Scripts` directory
3. Execute `python`
4. `import sys` then `sys.path`

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by scopatz** [comment]
How did you create the virtual environment?

**Comment by mwm-twx** [comment]
With `virtualenv xonshdir`. Followed by activating the environment and `python -m pip install xonsh`.

**Comment by scopatz** [comment]
Which shell are you activating the environment in xonsh, bash, etc? And which version of xonsh is in the virtualenv? Is it 0.9.21?

**Comment by mwm-twx** [comment]
I activated from cmd.exe; xonsh is 0.9.21.

**Comment by scopatz** [comment]
What is the value of PATH in the activated cmd session and in xonsh?

**Comment by mwm-twx** [comment]
No difference.

Cmd PATH:
```
C:\Users\someuser\AppData\Local\Programs\Python\Python38\Scripts\
C:\Users\someuser\AppData\Local\Programs\Python\Python38\
C:\Users\someuser\AppData\Roaming\Python\Python38\Scripts
```

Xonsh PATH:
```
EnvPath(
['C:\\Users\\someuser\\AppData\\Local\\Programs\\Python\\Python38\\Scripts\\',
 'C:\\Users\\someuser\\AppData\\Local\\Programs\\Python\\Python38\\',
 'C:\\Users\\someuser\\AppData\\Roaming\\Python\\Python38\\Scripts']
)
```

**Comment by scopatz** [comment]
Hmm this may be a xonsh bug. CC @melund?

**Comment by melund** [comment]
I haven't used virtualenv or standard python installations. Only anaconda. So I think I will need to find a clean machine to test this.

**Comment by mwm-twx** [comment]
It may be due to how the command is passed to POpen, based on the windows [CreateProcess docs](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa) :

"If lpApplicationName is NULL, the first white space–delimited token of the command line specifies the module name. ... If the file name does not contain an extension, .exe is appended. ... If the file name **does not contain a directory path**, the system searches for the executable file in the following sequence:
    The **directory from which the application loaded.**
     ..."

**Comment by melund** [comment]
@mwm-twx I think xonsh should handle these things, though.

Do you mind typing "where python" in a normal conhost/cmd.exe command prompt and post the output?

**Comment by mwm-twx** [comment]
I do not believe this to be an issue with `PATH`, as `which python` returns the correct interpreter.

As far as I can tell from looking at the code and `$XONSH_TRACE_SUBPROC`, the requested command `python` is passed to `Popen`; rather than xonsh performing a `which` equivalent and providing the resulting path to `Popen`.  On windows, this results in a call to `CreateProcess`, which will check _first_ for an executable with that name in the same location as the executable calling `CreateProcess`.  In the case of a virtualenv, there is a python executable in that directory, so it's always used.

I have tried copying the xonsh executable to another location, but that didn't help.  I think the only way for xonsh to get around this is to perform the search on `PATH` and use the full path in the `Popen` call.

**Comment by melund** [comment]
>I have tried copying the xonsh executable to another location, but that didn't help

I don't understand why that didn't help. If the executable is not the in current directory then CreateProcess should find it in $PATH...

**Comment by mwm-twx** [comment]
My theory was the executable generated was tied to the python executable in the virtualenv.  A bit of spelunking confirms this to be the case.

Running `strings` on xonsh.exe contains a reference to `C:\Users\Vinay\Projects\simple_launcher\dist\t64.pdb`, which indicates the executable is created from https://bitbucket.org/vinay.sajip/simple_launcher/src/master/.  The readme indicates the executable is created by concatenating a launcher executable with a python script, creating a shebang line to use the correct python.  The strings from the executable confirm this is what gets executed:

```python
#!c:\users\someuser\tools\xonsh\scripts\python.exe
__main__.py# -*- coding: utf-8 -*-
import re
import sys
from xonsh.main import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

The script is executed as a child process by the launcher (strings also includes `CreateProcessW`), so the final xonsh enviroment is the same as running `python -m xonsh`, regardless of where xonsh.exe is started from.

**Comment by anki-code** [comment]
The Windows CreateProcess directory-of-caller behavior was fixed in #4077. Closing as resolved. Please open new issue if you can reproduce on a current xonsh release.
